examples/dreambooth/train_dreambooth_lora.py

```diff
@@ -1291,15 +1268,17 @@ def compute_text_embeddings(prompt):
                     pipeline_args = {"prompt": args.validation_prompt}
 
                 if args.validation_images is None:
-                    images = [
-                        pipeline(**pipeline_args, generator=generator).images[0]
-                        for _ in range(args.num_validation_images)
-                    ]
+                    images = []
+                    for _ in range(args.num_validation_images):
+                        with torch.cuda.amp.autocast():
```

Comment:
Why these changes? We try to avoid running pipelines in autocast if possible

Reply:
I think I copy and pasted from the regular dreambooth training script which runs the validation inference under autocast. Will remove

Reply:
Actually I might have added this because when we load the rest of the model in fp16, we keep the lora weights in fp32 and needed autocast for them to work together

Reply:
Yes, looking through the commit history, that's why I added it. Is that ok?

Reply:
For running inference validation in intervals, I think keeping autocast is okay as it helps keep things simple.

Reply:
Why didn't we need it before?

Reply:
Ok so I was wrong that we needed it because of the difference in dtype in the lora weights. The lora layer casts manually internally.


The issue was the dtype of the output of the unet being passed to the vae.

The difference is that in the `main` version of the script, the unet is not wrapped in amp, so the output of the unet during validation is the same dtype as the unet, fp16. 

In the branch, the full unet is wrapped in amp so even though the unet is loaded in fp16, the output is fp32 and then there's an error when the fp32 ending latents are passed to the fp16 vae.

Reply:
The alternative to wrapping this section in amp is to put a check either in the pipeline or the beginning of the vae for the dtype of the initial latents and manually cast them if necessary. I like to prefer manual casts like that in case the caller expects the execution in the dtype of their input (which is a reasonable assumption imo). So I would prefer to leave the amp decorator in this case